Spreadsheet plugin docs (#265)
ckeditor/ckeditor4-docs: docs/features/spreadsheets/README.md

* Basic data styling like bold, italics, underline and so on.
* Defining data types and formatting with support for validation invalid data.
* Inserting formulas.
* Applying various conditional formatting rules to single/multiple cells, entire columns or spreadsheets.

## Inserting spreadsheet widget

Spreadsheet widget can be inserted via spreadsheet dialog or using Content Templates plugin with prepared spreadsheet templates. Below you can see simple 3x3 spreadsheet widget right after being inserted into editor.

{@img assets/img/spreadsheets1.png Spreadsheet plugin widget}

## Converting existing tables

If you already have tables in your content and plan to use Spreadsheet plugin there is no need to recreate entire table from scratch as you may simply convert them to spreadsheet widget instance:

{@img assets/img/spreadsheets2.png Spreadsheet plugin convert existing table context menu option}

## Sorting

Sorting is a must to efficiently analyze and present any tabular data, thus spreadsheet plugin allows for sorting data based on single or multiple columns:

{@img assets/img/spreadsheets3.png Spreadsheet plugin widget with multisort}

## Basic styles support

> While spreadsheets are mostly about data, its presentation is also very important. Basic styles like bold or italics comes handy when trying to highlight or make more visible some parts of data or single spreadsheet cells:

Reviewer: There is also justify and removeformat support but not sure if it's worth pointing this out / treat as overall basicstyles support :thinking:

{@img assets/img/spreadsheets4.png Spreadsheet plugin widget with basic styling}

## Data types, formats and validation

Data is the essence of every spreadsheet or data-grid element. This means it requires correct typing so for example formulas and conditional formatting knows how to handle it. Spreadsheet plugin provides few data types like Text, Number, Date, Time, Password and various formats so visually it can be presented in various ways. Additionally, each data type has a strict validation so it is clearly visible when something is not right with the data:

{@img assets/img/spreadsheets5.png Spreadsheet plugin with Cells Type and Format dialog}

## Formulas

> Formulas provide a way to calculate many different things and automate data processing in any spreadsheet. Just type `=` and the rest of the formula, easy as one two three!

Author: Makes sense 👍 Copied and linked :D

{@img assets/img/spreadsheets6.png Spreadsheet plugin with formula preview}

> ## Conditional formatting

Author: I was also thinking about it but don't have any nice idea for now. Maybe adding such formatting in sample could be enough?

> ## Conditional formatting

Reviewer: I seems easy when you thinking about it but then writing it down... :smile: Maybe let's just add some sentence like:

`You can even drag and drop conditional formatting rules changing its priority to easily create features like a color scale. Put your hands on the color scale example with our demo at <link to demo>`?

> Need to add some colors to your data? Conditional formatting allows to format data (any cell, column or entire table) based on its content. You may mark cells red when they are empty, make them green when theri value is above any threshold or blue if the cell value contains specific text.

Author: Yep, I have the similar feeling, it sometimes hard to write it the other way (non-personal). I will try to rephrase.

> Need to add some colors to your data? Conditional formatting allows to format data (any cell, column or entire table) based on its content. You may mark cells red when they are empty, make them green when theri value is above any threshold or blue if the cell value contains specific text.

Author: There is more than I have expected :D

> Need to add some colors to your data? Conditional formatting allows to format data (any cell, column or entire table) based on its content. You may mark cells red when they are empty, make them green when theri value is above any threshold or blue if the cell value contains specific text.

Author: I see the other docs also use this form, I will still consult with @AnnaTomanek if that's ok. cc @AnnaTomanek

> Need to add some colors to your data? Conditional formatting allows to format data (any cell, column or entire table) based on its content. You may mark cells red when they are empty, make them green when theri value is above any threshold or blue if the cell value contains specific text.

Reviewer 2: ```suggestion
Do you need to add some colors to your data? Conditional formatting allows formatting data (any cell, column or entire table) based on its content. You can mark cells red when they are empty, make them green when their value is above any threshold or blue if the cell value contains a specific text.
```

We actually prefer the "you" form to "we" form as it's more engaging and less patronizing; it's not "our software", it becomes more of "your implementation", if you get what I mean. Anyway, this one may sound a bit too informal, so I proposed a way to reword it.
